### DISCUSSION TITLE
更新日志 - 20230310

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#637] [Announcements]
# 系统：
1. 固件集成 kmod-inet-diag 内核模块 #611 #581 
2. 提高内核兼容性，以便安装 OpenWRT 预编译的内核模块，可自行在软件包中安装 iscsi，mpls，openvswitch 等驱动  （x86, rk33xx, rpi4等使用5.4内核的设备）

# 应用：
* QuickStart：
  1. 修复旁路由模式切换以后默认网关没有立即生效的问题
  2. 修复Docker的开关  #633
  3. 修复静态IP联网时，自动获取DNS选项依然可用的问题
  4. 网络向导：切换非旁路由联网模式时，如果之前旁路由模式关闭了DHCP服务，就显示一个重新打开DHCP服务的开关
  5. 网络向导：旁路由向导在配置完成时提醒用户后续操作，以及跳转按钮使用旁路由IP

* iStore：
  1. 增加了 kmod-ipt-socket，iptables-mod-socket 兼容模块 #581 #570

# 设备：
* RK3568系列： 提高千兆网口的性能（包括R5S，R68s等），测试自建 PPPOE 服务可以跑满千兆 #518 
* R4S/R66s：修复从某些高速SD卡启动以后，软重启失败问题（软件降级到低速卡） #565

**Comment by weiwei52o** [comment]
加油啊，好爱你们家的固件。从以前老论坛的时候，就支持你们好多年了

**Comment by q405812306** [comment]
我发现虽然是按照端口来绑定挂载存储，但是如果两块或两块以上外挂存储还是会错乱，还不如用uuid的绑定。pi4。固件版本是iStoreOS 21.02.3 2023022415

**Reply by jjm2473** [reply]
难道是用USB hub连接的？

**Reply by q405812306** [reply]
不是。。。刚刚看了一下，在已有1块外挂硬盘下（sda），在插上另一块，原有就变成sdb，新插上的是sdc。。。如果是重启一下的话就能恢复正常

**Reply by jjm2473** [reply]
你可能理解错了按端口挂载，这个功能是自动的，也就是插上硬盘，如果没配置过挂载点，那就按端口的位置自动挂载，挂载点是`/mnt/sata1-1`，`/mnt/usb1-1`之类的。

你说的"按照端口来绑定"是自己手动配置挂载点，这时候是用uuid才不会乱啊。

**Reply by q405812306** [reply]
好的，那我手动绑定一下uuid

**Comment by LemonSoda-RPG** [comment]
每次istore关机，外挂的3.5寸机械硬盘就会突然断电，听声音可以明显感觉出来，不知道是不是我自己的问题。

**Reply by jjm2473** [reply]
什么硬件

**Reply by LemonSoda-RPG** [reply]
> 什么硬件

j4125

**Reply by jjm2473** [reply]
x86的驱动都是linux官方的，不太可能出问题

**Reply by LemonSoda-RPG** [reply]
> x86的驱动都是linux官方的，不太可能出问题

我又下单了一个大功率的电源，等到了再看看情况

**Reply by LemonSoda-RPG** [reply]
> x86的驱动都是linux官方的，不太可能出问题

经过测试发现了，内置的关机应用没问题，可以正常关机，但是那个如果执行那个重启应用，硬盘就会突然断电

**Comment by qzsnitz** [comment]
<html>
<body>
<p dir="auto" style="box-sizing: border-box; margin-top: 0px !important; margin-bottom: 16px; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">ipv6获取到不能访问<br style="box-sizing: border-box;">ipv6测试结果</p><div id="user-content-your_ipv4" dir="auto" style="box-sizing: border-box; margin-bottom: 0px !important; color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

  | 你的公网 IPv4 地址是 103.172.41.188
-- | --


你的公网 IPv4 地址是 103.172.41.188<p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px;">你的运营商（ISP）是 AKARI-NETWORKS-AS-AP Akari Networks</p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px;">没有检测到 IPv6 地址<span> </span><a href="http://www.testipv6.com/faq_no_ipv6.html" rel="nofollow" style="box-sizing: border-box; background-color: transparent; color: var(--color-accent-fg); text-decoration: none;">[更多信息]</a></p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px;">你只接入了 IPv4 互联网，不能访问纯 IPv6 网站。</p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px;">可向运营商咨询如何使用 IPv6，实现最佳的网络性能。<span> </span><a href="http://www.testipv6.com/faq_no_ipv6.html" rel="nofollow" style="box-sizing: border-box; background-color: transparent; color: var(--color-accent-fg); text-decoration: none;">[更多信息]</a></p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px;">你的 DNS 服务器（可能由运营商提供）已经接入 IPv6 互联网了。</p></div>
</body>
</html>ipv6获取到不能访问
ipv6测试结果

 	你的公网 IPv4 地址是 103.172.41.179你的公网 IPv4 地址是 103.172.41.188
你的运营商（ISP）是 AKARI-NETWORKS-AS-AP Akari Networks

没有检测到 IPv6 地址 [[更多信息]](http://www.testipv6.com/faq_no_ipv6.html)

你只接入了 IPv4 互联网，不能访问纯 IPv6 网站。

可向运营商咨询如何使用 IPv6，实现最佳的网络性能。 [[更多信息]](http://www.testipv6.com/faq_no_ipv6.html)

你的 DNS 服务器（可能由运营商提供）已经接入 IPv6 互联网了。

**Reply by jjm2473** [reply]
默认禁止了ipv6的dns解析，在dhcp/dns的高级设置可以开启。如果ipv6地址都没有，那就不是这个问题了。

**Reply by ewap2021** [reply]
我也是这样，不知道哪里出了问题

**Reply by ewap2021** [reply]
解决没有？

**Reply by ewap2021** [reply]
我没有禁这个，还是一样。> 默认禁止了ipv6的dns解析，在dhcp/dns的高级设置可以开启。如果ipv6地址都没有，那就不是这个问题了。

**Reply by loocao** [reply]
我照网上改了好久，终于有了IPv6。。。

**Comment by socpio** [comment]
iStoreOS

 
系统 - 正在刷写…
正在刷写系统…
切勿关闭电源！ DO NOT POWER OFF THE DEVICE!
等待数分钟后即可尝试重新连接到路由。您可能需要更改计算机的 IP 地址以重新连接。
Loading 请稍候.

几秒钟，又回到iStoreOS主页，然后就没然后了。
何解？

**Reply by socpio** [reply]
忘说设备了，R4S_4G版本。

**Reply by jjm2473** [reply]
可能你的系统关闭程序太慢，刷机程序等了一段时间，发现ip已经能连上了，就跳到首页了。关键看r4s的状态灯有没有闪，闪的话表示还在刷机

**Reply by jjm2473** [reply]
还有，如果是手动升级固件，看看下载的固件是不是完整，检查hash值对不对得上

**Reply by socpio** [reply]
> 可能你的系统关闭程序太慢，刷机程序等了一段时间，发现ip已经能连上了，就跳到首页了。关键看r4s的状态灯有没有闪，闪的话表示还在刷机

r4s的pow红灯和sys绿灯一直常亮，wan和lan就有流量的正常绿闪，等了十几分钟也没反应，尝试三五次无果，只能拆TF卡重刷再恢复设置了，手动升级也是如此，不解

**Reply by jjm2473** [reply]
你确定不是自己把系统的升级功能搞坏了？建议备份配置以后恢复出厂设置，再试试能不能升级。

在线升级是会校验固件完整性的，所以下载的固件肯定没问题，没法升级只能是升级程序出问题了，如果要是所有人都有问题，应该收到很多反馈了，但现在只有你反馈，估计还是你的系统出问题了。

**Reply by socpio** [reply]
也许吧，这次我已经格式化升级了，下次系统更新时候再试试

**Comment by noshenxian** [comment]
添加个RTL8821ac的驱动呗

**Reply by jjm2473** [reply]
如果openwrt本身就有的驱动，可以去软件包那边安装

**Reply by jjm2473** [reply]
试试在软件包那边安装`kmod-rtl8821ae`，或者执行`opkg update; opkg install kmod-rtl8821ae`

**Comment by maplerian** [comment]
请求添加一个 AX210 驱动

**Reply by jjm2473** [reply]
x86的话，已经集成了iwlwifi驱动，可能就差firmware了，你把dmesg日志贴出来看看缺少什么firmware

**Reply by jjm2473** [reply]
开机后执行`dmesg | grep iwlwifi`应该就能看到了

**Reply by maplerian** [reply]
dmesg | grep iwlwifi 没有显示, 看错WiFi信号, 是 MT7921A-K

**Reply by maplerian** [reply]
[    9.139007] mt7921_common: Unknown symbol mt76_connac_mcu_alloc_sta_req (err -2)
[    9.415747] mt7921_common: Unknown symbol mt76_connac_mcu_alloc_sta_req (err -2)
[    9.551689] mt7921_common: Unknown symbol mt76_connac_mcu_alloc_sta_req (err -2)
[    9.566923] kmodloader: - mt7921-common - 0
[    9.566942] kmodloader: dependency not loaded mt7921-common
[    9.566960] kmodloader: - mt7921e - 1
[    9.566978] kmodloader: dependency not loaded mt7921-common
[    9.566997] kmodloader: - mt7921s - 1
[11650.087303] mt7921_common: Unknown symbol mt76_connac_mcu_alloc_sta_req (err -2)
[11650.251338] mt7921_common: Unknown symbol mt76_connac_mcu_alloc_sta_req (err -2)
[11650.270558] kmodloader: - mt7921-common - 0
[11650.270577] kmodloader: dependency not loaded mt7921-common
[11650.270595] kmodloader: - mt7921e - 1
[11650.270613] kmodloader: dependency not loaded mt7921-common
[11650.270632] kmodloader: - mt7921s - 1
[11650.503191] mt7921_common: Unknown symbol mt76_connac_mcu_alloc_sta_req (err -2)
[11650.651202] mt7921_common: Unknown symbol mt76_connac_mcu_alloc_sta_req (err -2)
[11650.666566] kmodloader: - mt7921-common - 0
[11650.666585] kmodloader: dependency not loaded mt7921-common
[11650.666603] kmodloader: - mt7921e - 1
[11650.666620] kmodloader: dependency not loaded mt7921-common
[11650.666640] kmodloader: - mt7921s - 1
[11650.903182] mt7921_common: Unknown symbol mt76_connac_mcu_alloc_sta_req (err -2)
[11651.091143] mt7921_common: Unknown symbol mt76_connac_mcu_alloc_sta_req (err -2)
[11651.106503] kmodloader: - mt7921-common - 0
[11651.106523] kmodloader: dependency not loaded mt7921-common
[11651.106542] kmodloader: - mt7921e - 1
[11651.106560] kmodloader: dependency not loaded mt7921-common
[11651.106580] kmodloader: - mt7921s - 1

**Reply by jjm2473** [reply]
mt7921已经集成到固件里了，是不是你安装了openwrt仓库里面的，那可能要恢复出厂设置了。

**Reply by maplerian** [reply]
恢复出产设置, 会保留宽度信息吗?

**Reply by jjm2473** [reply]
什么宽度信息？恢复出厂设置自然什么都不保留。或者你不保留配置升级也行

**Comment by yangkaiyv** [comment]
一直很期待一个功能，因为家里的宽带偶尔会跳到非公的IP 需要重拨才能解决。因此希望能添加一个类似ikuai的异常IP重播的功能。设备是R68s 非常感谢；)

**Reply by jjm2473** [reply]
可能找运营商还更快解决呢。写这个插件应该不难

**Comment by fanym2022** [comment]
什么时候能添加R6S的固件，谢谢

**Reply by jjm2473** [reply]
暂时还没厂商找我们适配，而且3588价格比较高，市场应该会比3568小

**Comment by nettentacle** [comment]
保持设置并保留当前配置 还不是勾不上

**Reply by jjm2473** [reply]
什么设备？使用的是不是默认主题？而且默认状态不是勾上的吗？

**Reply by jjm2473** [reply]
如果是bug，麻烦提交issue，描述清楚。

**Reply by nettentacle** [reply]
国伟r86s，n5105 cpu版本，是默认主题，保持设置并保留当前默认状态不是勾上的，且无法勾上

**Reply by jjm2473** [reply]
什么浏览器

**Reply by jjm2473** [reply]
我在电脑和手机上都测试一轮，没发现有什么问题

**Reply by jjm2473** [reply]
是不是用了深色主题或者不是argon主题？

**Reply by jjm2473** [reply]
修复了，其实只是复选框的颜色不明显

**Comment by Johnsnow-s** [comment]
web已无法进入，如何通过ssh如何升级固件

**Reply by jjm2473** [reply]
虽然可以用命令升级，但是解释起来太麻烦，不如恢复出厂设置`firstboot -y -r`

**Comment by ZJUprince** [comment]
升级完成后 沙盒模式按钮不见了  希望能保留配置  如何解决

**Comment by popalanluck** [comment]
r68s刷任意版本的istoreos，安装shellclash后，在使用tproxy模式时都会报错，
提示为：RTNETLINK answers: Address family not supported by protocol。

据说是固件编译的问题，编译时要先配置 
IP_MULTIPLE_TABLES=y
IPV6_MULTIPLE_TABLES=y
希望能早日解决这问题。

**Reply by jjm2473** [reply]
试试这个固件有没有问题 https://we.tl/t-F6F84Jyz6O
